### ISSUE TITLE
"\xFF" from ASCII-8BIT to UTF-8 - rails 3 branch

### ISSUE THREAD

**Issue by cheolhom** [jstorimer/delayed_paperclip#17]
I've been looking at source, but not sure if its with delayed job, or paperclip, or this gem.  I've been trying both delayed_jobs collectiveidea, and tobi.  Any help on where to look would be appreciated.

"\xFF" from ASCII-8BIT to UTF-8
/.rvm/gems/ruby-1.9.2-p0/gems/paperclip-2.3.3/lib/paperclip/storage.rb:197:in `to_file'
/.rvm/gems/ruby-1.9.2-p0/gems/paperclip-2.3.3/lib/paperclip/attachment.rb:209:in`reprocess!'
/.rvm/gems/ruby-1.9.2-p0/bundler/gems/delayed_paperclip-ddb7c1e85bf6/lib/delayed/jobs/delayed_paperclip_job.rb:4:in `block in perform'
/.rvm/gems/ruby-1.9.2-p0/bundler/gems/delayed_paperclip-ddb7c1e85bf6/lib/delayed/jobs/delayed_paperclip_job.rb:16:in`process_job'
/.rvm/gems/ruby-1.9.2-p0/bundler/gems/delayed_paperclip-ddb7c1e85bf6/lib/delayed/jobs/delayed_paperclip_job.rb:3:in `perform'

**Comment by jstorimer** [comment]
Did you find the answer to this? 

Based on the stacktrace it looks like an issue with Paperclip, though it's possible delayed_paperclip is causing the problem.

**Comment by cheolhom** [comment]
I was unable to resolve this,  I just went to using delayed_job, and paperclip without this gem.  thanks

**Comment by heijmerikx** [comment]
Same problem here, while trying to regenerate my thumbnails i get the error too.

**Comment by jstorimer** [comment]
I'm trying to reproduce this but it's working fine for me. It must have something to do with the image content you guys are working with. Can you share one of those images with me?

It would be great if you could also give concrete steps to reproduce the issue.

**Comment by heijmerikx** [comment]
I only get this issue when trying to regenerate the thumbnails, on my local dev environtment and on the server i've deployed my application to.

I'm even happy to let you access my server on where this is happening. If you would like to, drop me an email and i'll provide the credentials. If you're only interested in the image content, i'll provide the develepment URL.

**Comment by heijmerikx** [comment]
Great, i see it's fixed about 3 days ago. Thank you!

**Comment by seangaffney** [comment]
I ran into this same issue (Rails 3.0.3, Ruby 1.9.2) and found it to be an error with Paperclip while attempting to upload to S3. Within the Paperclip gem itself, it creates a Tempfile which is w+ by default, always, not wb+ (binary) like the S3 object expects.  I fixed it by adding a line after the Tempfile is created to convert it to binary using the .binmode method.

File in question is in the Paperclip gem.
{gemdir}/gems/paperclip-2.3.5/lib/paperclip//storage/s3.rb

Take a look at the gist for more info:
https://gist.github.com/730724

Not sure why it works without this gem and breaks with it, but the above solved the issue for me. Hope it helps someone else!
